processor/bloom.go

```diff
@@ -0,0 +1,24 @@
+package processor
+
+// Prime number less than 256
```

Comment:
only comment I have is that could you put WHY a prime, and WHY below 256. Because I am going to scratch my head at that sometime in the future.

Reply:
That probably doesn't strictly need to be a prime number anymore, but it seemed to work reasonably well so I left it. 

Since ASCII characters aren't well distributed in the 0-255 range, I was originally trying to do something with multiplying by a large prime and modulus down again to "scatter" the values better. I eventually realised that multiplying the number by itself worked better for this purpose - the xor was thrown in to deal with small numbers having small squares.

Less than 256 was because I was originally going to just increase the size to `uint16` until I realised I was going to need a lot more bits. The thinking was that the maximum byte (255) multiplied by the prime wouldn't overflow the `uint16`.

Ultimately, the whole thing is just based on my own maths not a well known hashing algorithm. I was trying to keep the calculation lightweight since it was going to be done in a tight loop - but with the addition of a prebuilt lookup table that's not such a problem anymore.

Coming back to it after sleeping on it, we could probably replace the arbitrary prime + multiplication with `math/rand`: `rand.New(rand.NewSource(int64(b))).Uint64()`. It might even produce more random results and function better. I'll put in a new PR to change it out so it doesn't lead to confusion in future (and fill out some more comments to explain things better).